presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -359,43 +359,46 @@ public void recordHeartbeat()
     public void pruneInfo()
     {
         QueryInfo queryInfo = finalQueryInfo.get();
-        if (queryInfo == null || queryInfo.getOutputStage() == null) {
```

Comment:
Leave the check for `getOutputStage()` here using `isPresent()`, then the rest of the code can stay the same.